mparray/special/_special.py

```diff
@@ -31,22 +31,27 @@
 
 
 @vectorize
-def gammainc(x, a):
-    return mp.gammainc(x, a=0, b=a, regularized=True)
+def gammainc(a, x):
+    return mp.gammainc(a, a=0, b=x, regularized=True)
 
 
 @vectorize
-def gammaincc(x, a):
-    return mp.gammainc(x, a=a, b=mp.inf, regularized=True)
+def gammaincc(a, x):
+    return mp.gammainc(a, a=x, b=mp.inf, regularized=True)
 
 
 @vectorize
 def ndtri(x):
+    if x == 0:
+        return -mp.inf
+    if x == 1:
+        return mp.inf
+    if x < 0 or x > 1:
+        return mp.nan
+
     extra_dps = int(mp.ceil(-mp.log10(x)))
-    mp.dps += extra_dps
-    res = mp.sqrt(2) * mp.erfinv(2 * x - mp.one)
-    mp.dps -= extra_dps
-    return res
+    with mp.workdps(mp.dps + extra_dps):
```

Comment:
Cool. I thought I looked for something like this but I guess I didn't look carefully enough.